### PR TITLE
Add contour and contourf capabilities for non map plots

### PR DESCRIPTION
**Pull request by kevindougherty-noaa**
Adding contour and contourf capabilities to plotting to be able to plot things such as cross sections/Hovmoller/etc.

### CODE REVIEW

**Comment by CoryMartin-NOAA** [comment]
Can you upload the test figures here just to make sure we have record it is working, @kevindougherty-noaa ? Otherwise the code looks good to me and thanks for adding this!

**Comment by kevindougherty-noaa** [comment]
image

Sure thing @CoryMartin-NOAA

File: src/emcpy/plots/plots.py [hunk 1]
```diff
@@ -175,6 +175,68 @@ def __init__(self, x, y, z):
...
+        self.vmin = None
+        self.vmax = None
+        self.origin = None
+        self.extent = None
+        self.locator = None
+        self.extend = None
+        self.colorbar = False
+
+
+class ContourfPlot:
```
Comment (CoryMartin-NOAA) [inline src/emcpy/plots/plots.py]: Curious to hear thoughts from @EdwardSafford-NOAA, @ADCollard or @JessicaMeixner-NOAA , what about "FilledContourPlot"? "Contourf" might not be clear to non-matplotlib experts.

Comment (JessicaMeixner-NOAA) [reply src/emcpy/plots/plots.py]: Countourf is also used in matlab, so while i'm not a matplotlib expert, it was easy to know what it was.  That being said, "FilledCountourPlot" is  more obvious.

**Review by EdwardSafford-NOAA — DISMISSED** [review]
Looks good to me, though I vote for FilledContourPlot too.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Pull request merged**